homeassistant/components/sensor/pyload.py

```diff
@@ -0,0 +1,170 @@
+"""
+Support for monitoring pyLoad download speed.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.pyload/
+"""
+import logging
+from datetime import timedelta
+
+import requests
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    CONF_HOST, CONF_PASSWORD, CONF_USERNAME, CONF_NAME, CONF_PORT,
+    CONF_SSL, CONTENT_TYPE_JSON, CONF_MONITORED_VARIABLES)
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'pyLoad'
+DEFAULT_PORT = 8000
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=5)
+
+SENSOR_TYPES = {
+    'speed': ['speed', 'Speed', 'MB/s'],
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_MONITORED_VARIABLES, default=['speed']):
+        vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)]),
```

Comment:
There is only one sensor type possible. If there are no other sensors possible with pyload then `speed` should be set and configuration option be removed. If there are other you plan to integrate then it's ok.